### PR TITLE
ECSql Query Blob Abbreviation Support

### PR DESCRIPTION
**Pull request by Aurora-Lane**
Queries now have an optional parameter to abbreviate any blob value returned, aptly named abbreviateBlobs. Passing true 
to this value will truncate any blob in the query to 1 byte instead of returning its entire contents.

Corresponding imodel02 pr: https://dev.azure.com/bentleycs/iModelTechnologies/_git/imodel02/pullrequest/134867

### CODE REVIEW

File: core/backend/src/ECDb.ts [hunk 1]
```diff
@@ -322,7 +322,7 @@ export class ECDb implements IDisposable {
    * See [ECSQL row format]($docs/learning/ECSQLRowFormat) for details about the format of the returned rows.
    * @internal
    */
-  public async queryRows(ecsql: string, bindings?: any[] | object, limit?: QueryLimit, quota?: QueryQuota, priority?: QueryPriority, restartToken?: string): Promise<QueryResponse> {
+  public async queryRows(ecsql: string, bindings?: any[] | object, limit?: QueryLimit, quota?: QueryQuota, priority?: QueryPriority, restartToken?: string, abbreviateBlobs?: boolean): Promise<QueryResponse> {
```
Comment (pmconne) [inline core/backend/src/ECDb.ts:325]: 6 optional arguments is unwieldy - did you consider bundling them up into an interface?

**Review by pmconne — CHANGES_REQUESTED** [review]

**Comment by pmconne** [comment]
@Aurora-Lane this seems like it would be particularly useful to frontend code (reduce size of QueryResponses transmitted over network by abbreviating blobs) - is support for that planned?

**Comment by Aurora-Lane** [comment]
> @Aurora-Lane this seems like it would be particularly useful to frontend code (reduce size of QueryResponses transmitted over network by abbreviating blobs) - is support for that planned?

Yup, that's why this pr in particular is still a draft - I'm still looking at implementing this into rpc and frontend as cleanly as possible

File: core/backend/src/ECDb.ts (hunk 1 above)
Comment (Aurora-Lane) [reply core/backend/src/ECDb.ts:325]: looked at this - an interface would definitely be a more optimal solution but will break more things dependent on the backend. It's on the todo list for the future, but it's not in scope for this PR

**Review by pmconne — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**